contracts/CodexRecordMetadata.sol

```diff
@@ -38,29 +37,21 @@ contract CodexRecordMetadata is ERC721Token {
    * @param _newNameHash bytes32 The new sha3 hash of the name
    * @param _newDescriptionHash bytes32 The new sha3 hash of the description
    * @param _newFileHashes bytes32[] The new sha3 hashes of the files associated with the token
-   * @param _providerId (optional) string An ID that identifies which provider is
-   *  minting this token
-   * @param _providerMetadataId (optional) string An arbitrary provider-defined ID that
-   *  identifies the metadata record stored by the provider
+   * @param _data (optional) bytes Additional data that will be emitted with the Modified event
    */
   function modifyMetadataHashes(
     uint256 _tokenId,
     bytes32 _newNameHash,
     bytes32 _newDescriptionHash,
     bytes32[] _newFileHashes,
-    string _providerId, // TODO: convert to bytes32?
-    string _providerMetadataId  // TODO: convert to bytes32?
+    bytes _data
   )
     public
     onlyOwnerOf(_tokenId)
   {
     // nameHash is only overridden if it's not a blank string, since name is a
     //  required value
-    //
-    // NOTE: is this the best way to check for an empty bytes32 array?
-    //  would (_newNameHash != "") be better in any way?
-    //  see: https://ethereum.stackexchange.com/questions/27227/why-does-require-length-of-bytes32-0-not-work
-    if (_newNameHash[0] != 0) {
+    if (_newNameHash[0] != 0x0) {
```

Comment:
Are `0` and `0x0` not equivalent?

Reply:
I also think keeping the link to the stackexchange issue might be useful since I don't think it's entirely obvious why we're doing `_newNameHash[0] != 0x0`.

Reply:
They are equivalent, but I actually think this is more clear that we're checking for a null byte and not an integer value. I can add a comment here for context.

Reply:
Ah yeah, that's a fair point.